lib/common/strings.c

```diff
@@ -894,24 +894,23 @@ pcmk__str_in_list(GList *lst, const gchar *s, uint32_t flags)
 }
 
 static bool
-str_any_of(bool casei, const char *s, va_list args)
+str_any_of(const char *s, va_list args, uint32_t flags)
 {
-    bool rc = false;
+    if (pcmk_is_set(flags, pcmk__str_null_matches) && s == NULL) {
+        return true;
+    }
 
-    if (s != NULL) {
-        while (1) {
-            const char *ele = va_arg(args, const char *);
-
-            if (ele == NULL) {
-                break;
-            } else if (pcmk__str_eq(s, ele,
-                                    casei? pcmk__str_casei : pcmk__str_none)) {
-                rc = true;
-                break;
-            }
+    while (1) {
+        const char *ele = va_arg(args, const char *);
+
+        if (ele == NULL) {
+            break;
+        } else if (pcmk__str_eq(s, ele, flags)) {
+            return true;
         }
     }
-    return rc;
+
+    return false;
```

Comment:
This should work fine, but if s==NULL without null_matches we'll unnecessarily run through the list -- should be trivial to do if s==NULL return is_set null_matches.

Reply:
Fixed.  I'll wait to push again until after I have to rebase.